charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -53,8 +63,14 @@ logs:
 {{- $_ := set $config.service.pipelines.metrics "receivers" (mustWithout $config.service.pipelines.metrics.receivers "prometheus") }}
 {{- $_ := unset $config.service.telemetry.metrics "address" }}
```

Comment:
The logic in the IF above also needs adjusted to unset the new default prometheus reader when internalTelemetryViaOTLP.metrics is enabled

Reply:
When I generate the examples for internalTelemetryViaOTLP i don't get any changes in the example. 

Shouldn't this be the default behavior? 

Do you mean: `internalTelemetryViaOTLP.metrics.enabled` and `.Values.config.service.telemetry.metrics.address` should be working on the same time?

Reply:
I mean that when `internalTelemetryViaOTLP.metrics.enabled=true` that 

```yaml
  service:
    telemetry:
      metrics:
        readers:
          - pull:
              exporter:
                prometheus:
                  host: ${env:MY_POD_IP}
                  port: 8888
```
should be replaced by the otlp exporter for metrics. In the generated example you can see the otlp export is now missing.